Non-embeddable compiler JAR compilations are broken in 1.0.7

After updating KSP from `1.7.20-1.0.6` to `1.7.20-1.0.7` projects with the following gradle property
```properties
kotlin.native.useEmbeddableCompilerJar=false
```
fail to build in the `kspKotlin` tasks with
```
e: java.lang.IllegalStateException: The provided plugin com.google.devtools.ksp.KotlinSymbolProcessingComponentRegistrar is not compatible with this version of compiler
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment$Companion.registerExtensionsFromPlugins$cli(KotlinCoreEnvironment.kt:664)
	...
Caused by: java.lang.AbstractMethodError: Receiver class com.google.devtools.ksp.KotlinSymbolProcessingComponentRegistrar does not define or inherit an implementation of the resolved method 'abstract void registerProjectComponents(com.intellij.mock.MockProject, org.jetbrains.kotlin.config.CompilerConfiguration)' of interface org.jetbrains.kotlin.compiler.plugin.ComponentRegistrar.
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment$Companion.registerExtensionsFromPlugins$cli(KotlinCoreEnvironment.kt:656)
```

Looking at the changes between those versions I think this is caused by https://github.com/google/ksp/commit/f35af30f0aea18b9cf6ce84f22af7bf2f2df34ab.
The `symbol-processing` dependency seems to be added regardless of the `useEmbeddableCompilerJar` setting.

--- a/gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt
+++ b/gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt
@@ -78,8 +78,10 @@ class KspGradleSubplugin @Inject internal constructor(private val registry: Tool
         const val KSP_PLUGIN_ID = "com.google.devtools.ksp.symbol-processing"
         const val KSP_API_ID = "symbol-processing-api"
         const val KSP_COMPILER_PLUGIN_ID = "symbol-processing"
+        const val KSP_COMPILER_PLUGIN_ID_NON_EMBEDDABLE = "symbol-processing-cmdline"
         const val KSP_GROUP_ID = "com.google.devtools.ksp"
         const val KSP_PLUGIN_CLASSPATH_CONFIGURATION_NAME = "kspPluginClasspath"
+        const val KSP_PLUGIN_CLASSPATH_CONFIGURATION_NAME_NON_EMBEDDABLE = "kspPluginClasspathNonEmbeddable"
 
         @JvmStatic
         fun getKspOutputDir(project: Project, sourceSetName: String, target: String) =
@@ -228,8 +230,19 @@ class KspGradleSubplugin @Inject internal constructor(private val registry: Tool
         )
         project.dependencies.add(
             KSP_PLUGIN_CLASSPATH_CONFIGURATION_NAME,
-
             "$KSP_GROUP_ID:$KSP_COMPILER_PLUGIN_ID:$KSP_VERSION"
+        )
+
+        val kspClasspathCfgNonEmbeddable = project.configurations.maybeCreate(
+            KSP_PLUGIN_CLASSPATH_CONFIGURATION_NAME_NON_EMBEDDABLE
+        )
+        project.dependencies.add(
+            KSP_PLUGIN_CLASSPATH_CONFIGURATION_NAME_NON_EMBEDDABLE,
+            "$KSP_GROUP_ID:$KSP_API_ID:$KSP_VERSION"
+        )
+        project.dependencies.add(
+            KSP_PLUGIN_CLASSPATH_CONFIGURATION_NAME_NON_EMBEDDABLE,
+            "$KSP_GROUP_ID:$KSP_COMPILER_PLUGIN_ID_NON_EMBEDDABLE:$KSP_VERSION"
         )
 
         findJavaTaskForKotlinCompilation(kotlinCompilation)?.configure { javaCompile ->
@@ -473,12 +486,19 @@ class KspGradleSubplugin @Inject internal constructor(private val registry: Tool
                         configureAsKspTask(kspTask, false)
                         configureAsAbstractKotlinCompileTool(kspTask)
 
+                        val useEmbeddable = project.findProperty("kotlin.native.useEmbeddableCompilerJar")
+                            ?.toString()?.toBoolean() ?: true
+                        val classpathCfg = if (useEmbeddable) {
+                            kspClasspathCfg
+                        } else {
+                            kspClasspathCfgNonEmbeddable
+                        }
                         // KotlinNativeCompile computes -Xplugin=... from compilerPluginClasspath.
                         if (kspExtension.blockOtherCompilerPlugins) {
-                            kspTask.compilerPluginClasspath = kspClasspathCfg
+                            kspTask.compilerPluginClasspath = classpathCfg
                         } else {
                             kspTask.compilerPluginClasspath =
-                                kspClasspathCfg + kotlinCompileTask.compilerPluginClasspath!!
+                                classpathCfg + kotlinCompileTask.compilerPluginClasspath!!
                             kspTask.compilerPluginOptions.addPluginArgument(kotlinCompileTask.compilerPluginOptions)
                         }
                         kspTask.commonSources.from(kotlinCompileTask.commonSources)
@@ -540,7 +560,7 @@ class KspGradleSubplugin @Inject internal constructor(private val registry: Tool
     override fun getPluginArtifactForNative(): SubpluginArtifact? =
         SubpluginArtifact(
             groupId = "com.google.devtools.ksp",
-            artifactId = KSP_COMPILER_PLUGIN_ID,
+            artifactId = KSP_COMPILER_PLUGIN_ID_NON_EMBEDDABLE,
             version = KSP_VERSION
         )
 }

--- a/integration-tests/src/test/kotlin/com/google/devtools/ksp/test/KMPImplementedIT.kt
+++ b/integration-tests/src/test/kotlin/com/google/devtools/ksp/test/KMPImplementedIT.kt
@@ -222,6 +222,29 @@ class KMPImplementedIT {
     }
 
     @Test
+    fun testNonEmbeddableArtifact() {
+        Assume.assumeFalse(System.getProperty("os.name").startsWith("Windows", ignoreCase = true))
+        val gradleRunner = GradleRunner.create().withProjectDir(project.root)
+
+        gradleRunner.withArguments(
+            "--configuration-cache-problems=warn",
+            "-Pkotlin.native.useEmbeddableCompilerJar=false",
+            ":workload-linuxX64:kspTestKotlinLinuxX64"
+        ).build()
+
+        gradleRunner.withArguments(
+            "--configuration-cache-problems=warn",
+            "-Pkotlin.native.useEmbeddableCompilerJar=true",
+            ":workload-linuxX64:kspTestKotlinLinuxX64"
+        ).build()
+
+        gradleRunner.withArguments(
+            "--configuration-cache-problems=warn",
+            ":workload-linuxX64:kspTestKotlinLinuxX64"
+        ).build()
+    }
+
+    @Test
     fun testLinuxX64ErrorLog() {
         Assume.assumeFalse(System.getProperty("os.name").startsWith("Windows", ignoreCase = true))
         val gradleRunner = GradleRunner.create().withProjectDir(project.root)

